library/model/book.py

```diff
@@ -37,109 +50,82 @@ def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn, typ
         self.borrowed_items = borrowed_items
 
     @classmethod
-    def from_borrowed_book(cls, borrowed_book: "BorrowedBook") -> "Book":
+    def from_borrowed_book(cls, borrowed_book: "BookCopy") -> "Book":
         book = Book(
-            borrowed_book.title,
-            borrowed_book.authors,
-            borrowed_book.publisher,
-            borrowed_book.publication_date,
-            borrowed_book.genres,
-            borrowed_book.pages,
             borrowed_book.isbn,
-            borrowed_book._book_type,
-            borrowed_book.duration,
-            borrowed_book.existing_items,
-            borrowed_book.borrowed_items,
+            borrowed_book._book_type
+            borrowed_book.genres
         )
         return book
 
     def can_borrow(self) -> bool:
         if self._book_type == "Paper":
             return self.existing_items - self.borrowed_items > 0
-        elif self._book_type == "Electronic":
-            return True
-        elif self._book_type == "Audio":
+        elif self._book_type == "Electronic" or self._book_type == "Audio":
             return True
         else:
             raise AttributeError("No such book type...")
 
     def get_approximate_duration(self) -> int:
-        if self._book_type == "Paper":
-            return self.pages * 3 * 60
-        elif self._book_type == "Electronic":
-            return self.pages * 5 * 60
-        elif self._book_type == "Audio":
-            return self.duration
+        if self._book_type in self._booktype_duration.keys():
+            return self._booktype_duration[self._book_type]
         else:
             raise AttributeError("No such book type...")
 
     def get_weekly_fee(self) -> int:
-        if self._book_type == "Paper":
-            return 5
-        elif self._book_type == "Electronic":
-            return 2
-        elif self._book_type == "Audio":
-            return 2
+        if(self._book_type in self._booktype_fee.keys()):
+            return self._booktype_fee[self._book_type]
         else:
             raise AttributeError("No such book type...")
```

Comment:
Redundant, no? A failed dict lookup will raise a `KeyError`. Maybe you can let it propagate.